packages/react/src/profile/useCreateProfile.ts

```diff
@@ -1,19 +1,36 @@
-import { DuplicatedHandleError } from '@lens-protocol/domain/use-cases/profile';
+import { TransactionKind } from '@lens-protocol/domain/entities';
+import {
+  CreateProfileRequest,
+  DuplicatedHandleError,
+} from '@lens-protocol/domain/use-cases/profile';
 import { useState } from 'react';
 
+import {
+  TransactionState,
+  useHasPendingTransaction,
+} from '../transactions/adapters/TransactionQueuePresenter';
 import { useCreateProfileController } from './adapters/useCreateProfileController';
 
 export function useCreateProfile() {
   const [error, setError] = useState<DuplicatedHandleError | null>(null);
-  const [isPending, setIsPending] = useState(false);
+  const [isPending, setIsPending] = useState<boolean>(false);
+  const [handle, setHandle] = useState<string>('');
   const createProfile = useCreateProfileController();
 
-  return {
-    create: async (handle: string) => {
-      setIsPending(true);
+  const hasPendingTx = useHasPendingTransaction(
+    (transaction): transaction is TransactionState<CreateProfileRequest> =>
+      transaction.request.kind === TransactionKind.CREATE_PROFILE &&
+      transaction.request.handle === handle,
+  );
 
+  return {
+    create: async (_handle: string) => {
       try {
-        const result = await createProfile(handle);
+        setHandle(_handle);
+        setIsPending(true);
```

Comment:
We need this extra state here to achieve the "instant loading" feel in the UI.

Without this, you click the `Create Profile` button and the `isPending` flag remains as `false` for a few ms/s, giving the sense to the user that nothing is happening.